backend/src/main/java/codezap/global/logger/RequestResponseLogger.java

```diff
@@ -31,29 +40,82 @@ protected void doFilterInternal(HttpServletRequest request, HttpServletResponse
+                "[Request] %s %s, 헤더 값: %s",
+                request.getMethod(),
+                request.getRequestURI(),
+                getHeaderAsJson(requestWrapper));
+        String responseMessage = String.format(
+                "[Response] status: %d, duration: %dms, headers: %s",
+                status,
+                duration,
+                getHeaderAsJson(responseWrapper));
+
+        logError(status, requestMessage, responseMessage);
+        logWarn(status, requestMessage, responseMessage);
+        logInfo(status, requestMessage, responseMessage);
+    }
+
+    private String getHeaderAsJson(ContentCachingRequestWrapper requestWrapper) {
+        Map<String, String> headersMap = new HashMap<>();
         requestWrapper.getHeaderNames().asIterator().forEachRemaining(headerName -> {
             String headerValue = requestWrapper.getHeader(headerName);
-            headerAndValue.append(headerName).append(" : ").append(headerValue).append("\n");
+            headersMap.put(headerName, headerValue);
         });
 
-        return headerAndValue.toString();
+        return convertMapToJson(headersMap);
     }
 
-    private String getHeaderAndValue(ContentCachingResponseWrapper requestWrapper) {
-        return requestWrapper.getHeaderNames().stream().map(headerName -> {
-            String headerValue = requestWrapper.getHeader(headerName);
-            return headerName + " : " + headerValue;
-        }).collect(Collectors.joining("\n"));
+    private String getHeaderAsJson(ContentCachingResponseWrapper responseWrapper) {
+        Map<String, String> headersMap = new HashMap<>();
+        responseWrapper.getHeaderNames().forEach(headerName -> {
+            String headerValue = responseWrapper.getHeader(headerName);
+            headersMap.put(headerName, headerValue);
+        });
+
+        return convertMapToJson(headersMap);
+    }
+
+    private String convertMapToJson(Map<String, String> map) {
+        try {
+            return objectMapper.writeValueAsString(map);
+        } catch (JsonProcessingException e) {
+            log.error("헤더 정보를 JSON으로 변환하는 중 오류 발생", e);
+            return "{}";
+        }
+    }
+
+    private void logError(int status, String requestMessage, String responseMessage) {
+        boolean isError = status >= ERROR_CODE;
+        if (isError) {
+            log.error(requestMessage);
+            log.error(responseMessage);
+        }
+    }
+
+    private void logWarn(int status, String requestMessage, String responseMessage) {
+        boolean isWarn =  status >= WARN_CODE && status < ERROR_CODE;
+        if (isWarn) {
+            log.warn(requestMessage);
+            log.warn(responseMessage);
+        }
+    }
+
+    private void logInfo(int status, String requestMessage, String responseMessage) {
+        boolean isInfo = status < WARN_CODE;
+        if (isInfo) {
+            log.info(requestMessage);
+            log.info(responseMessage);
+        }
     }
```

Comment:
저는 다음과 같이 작성하는 게 더 낫다고 생각해서 제안해요~

이렇게하면 
1. Early Return으로 호출되는 코드 수를 줄일 수 있어요. 
2. `logResponse()` 메서드에서 하나의 메서드만 호출하면 돼요. 
3. `RequestResponseLogger` 클래스의 전체 라인 수가 줄어들어요. 

```suggestion
    private void logByStatus(int status, String requestMessage, String responseMessage) {
        if (status >= ERROR_CODE) {
            log.error(requestMessage);
            log.error(responseMessage);
            return;
        }
        if (status >= WARN_CODE) {
            log.warn(requestMessage);
            log.warn(responseMessage);
            return;
        }
        log.info(requestMessage);
        log.info(responseMessage);
    }
```